crates/events/src/enclave_event/compute_request/trbfv.rs

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+//
+// This file is provided WITHOUT ANY WARRANTY;
+// without even the implied warranty of MERCHANTABILITY
+// or FITNESS FOR A PARTICULAR PURPOSE.
+
+use serde::{Deserialize, Serialize};
+
+/// TrBFV modules defining the API for multithreaded compute
+/// Each module defines the event payloads that make up a compute request
+/// Each compute request should live independently and be self contained
+
+/// Input format for TrBFVRequest
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub enum TrBFVRequest {
+    GenEsiSss(gen_esi_sss::Request),
+    GenPkShareAndSkSss(gen_pk_share_and_sk_sss::Request),
+    GenDecryptionKey(gen_decryption_key::Request),
+    GenDecryptionShare(gen_decryption_share::Request),
+    ThresholdDecrypt(threshold_decrypt::Request),
+}
+
+/// Result format for TrBFVResponse
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub enum TrBFVResponse {
+    GenEsiSss(gen_esi_sss::Response),
+    GenPkShareAndSkSss(gen_pk_share_and_sk_sss::Response),
+    GenDecryptionKey(gen_decryption_key::Response),
+    GenDecryptionShare(gen_decryption_share::Response),
+    ThresholdDecrypt(threshold_decrypt::Response),
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub enum TrBFVError {
+    // Add errors here as required
+}
+
+pub mod gen_esi_sss {
+    /// This module defines event payloads that will generate esi smudging noise shamir shares to be shared with other members of the committee.
+    /// This has been separated from the general setup in order to be able to take advantage of parallelism
+    use crate::bytes::Bytes;
+    use e3_crypto::SensitiveBytes;
+    use serde::{Deserialize, Serialize};
+
+    #[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+    pub struct Request {
+        /// BFV Params
+        params: Bytes,
+        /// Max number of ciphertexts
+        max_num_ciphertexts: u64,
+    }
+
+    #[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+    pub struct Response {
+        /// The smudging noise shares
+        esi_sss: Vec<SensitiveBytes>,
+    }
+}
+
+pub mod gen_pk_share_and_sk_sss {
+    /// This module defines event payloads that will generate the public key share as well as the sk shamir secret shares to be distributed to other members of the committee.
+    /// This has been separated from the esi setup in order to be able to take advantage of parallelism
+    use crate::bytes::Bytes;
+    use e3_crypto::SensitiveBytes;
+    use serde::{Deserialize, Serialize};
+
+    #[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+    pub struct Request {
+        /// BFV Params
+        params: Bytes,
+    }
```

Comment:
Same here, we're missing the party info
I guess we could have a `PartyInfo` struct 

```
pub struct PartyInfo {
    pub party_id: u32,
    pub num_parties: u32,
    pub threshold: u32,
}
```

Reply:
We actually don't need the same in both from what I can tell.

Reply:
Smudging noise only needs max num ciphertexts but here we need threshold etc.